server/controllers/admin/overview.ts

```diff
@@ -0,0 +1,42 @@
+import { Request, Response, NextFunction } from 'express';
+import {
+  pharmacyCount,
+  productCount,
+  requestsCount,
+  pendingApplicationsCount,
+  openedApplicationsCount,
+  closedApplicationsCount,
+  expiredAndInStockProductsCount,
+} from '../../queries/admin/overview';
+
+const getAdminOverview = async (req:Request, res:Response, next:NextFunction) => {
+  const pharmaciesNumber = await pharmacyCount();
+  const productsNumber = await productCount();
+  const productsRequestsNumber = await requestsCount();
+
+  const pendingApplications = await pendingApplicationsCount();
+  const pendingApplicationsNumber = ((pendingApplications / pharmaciesNumber) * 100).toFixed(1);
+
+  const openedApplications = await openedApplicationsCount();
+  const openedApplicationsNumber = ((openedApplications / pharmaciesNumber) * 100).toFixed(1);
+
+  const closedApplications = await closedApplicationsCount();
+  const closedApplicationsNumber = ((closedApplications / pharmaciesNumber) * 100).toFixed(1);
+
+  const allKindProductsCount= await expiredAndInStockProductsCount();
```

Comment:
Since they are not independent of each other, so why await for one to do the other, use Promise.all to process them at the same time